addons/account/models/account_report.py

```diff
@@ -283,19 +283,21 @@ def copy(self, default=None):
         for old_report, new_report in zip(self, new_reports):
             code_mapping = {}
             for line in old_report.line_ids.filtered(lambda x: not x.parent_id):
-                line._copy_hierarchy(new_report, code_mapping=code_mapping)
-
-            # Replace line codes by their copy in aggregation formulas
-            for expression in new_report.line_ids.expression_ids:
-                if expression.engine == 'aggregation':
-                    copied_formula = f" {expression.formula} "  # Add spaces so that the lookahead/lookbehind of the regex can work (we can't do a | in those)
-                    for old_code, new_code in code_mapping.items():
-                        copied_formula = re.sub(f"(?<=\\W){old_code}(?=\\W)", new_code, copied_formula)
-                    expression.formula = copied_formula.strip()  # Remove the spaces introduced for lookahead/lookbehind
+                line.with_context(code_mapping=code_mapping).copy({'report_id': new_report.id})
 
+            new_report.update_aggregation_expression_formula(new_report.line_ids.expression_ids, code_mapping)
             old_report.column_ids.copy({'report_id': new_report.id})
         return new_reports
 
+    def update_aggregation_expression_formula(self, expression_ids, code_mapping):
+        # Replace line codes by their copy in aggregation formulas
+        for expression in expression_ids:
+            if expression.engine == 'aggregation':
```

Comment:
```suggestion
            if expression.engine != 'aggregation':
                continue
```
This would improve the readablitiy and less code indents